src/cmd/linuxkit/run_vcenter.go

```diff
@@ -0,0 +1,300 @@
+	if *newVM.path == "" {
+		log.Fatalf("No file specified")
+	}
+	dsurl := dss.NewURL(fmt.Sprintf("%s/%s", *newVM.vmName, fileName))
+
+	p := soap.DefaultUpload
+	if err := c.Client.UploadFile(*newVM.path, dsurl, &p); err != nil {
+		log.Fatalf("Unable to upload file to vCenter Datastore\n%v", err)
+	}
+}
+
+func addNIC(ctx context.Context, vm *object.VirtualMachine, net object.NetworkReference) {
+	backing, err := net.EthernetCardBackingInfo(ctx)
+	if err != nil {
+		log.Fatalf("Unable to determine vCenter network backend\n%v", err)
+	}
+
+	netdev, err := object.EthernetCardTypes().CreateEthernetCard("vmxnet3", backing)
+	if err != nil {
+		log.Fatalf("Unable to create vmxnet3 network interface\n%v", err)
+	}
+
+	log.Infof("Adding VM Networking")
+	var add []types.BaseVirtualDevice
+	add = append(add, netdev)
+
+	if vm.AddDevice(ctx, add...); err != nil {
+		log.Fatalf("Unable to add new networking device to VM configuration\n%v", err)
+	}
+}
+
+func addVMDK(ctx context.Context, vm *object.VirtualMachine, dss *object.Datastore, newVM vmConfig) {
+	devices, err := vm.Device(ctx)
+	if err != nil {
+		log.Fatalf("Unable to read devices from VM configuration\n%v", err)
+	}
+
+	controller, err := devices.FindDiskController("scsi")
+	if err != nil {
+		log.Fatalf("Unable to find SCSI device from VM configuration\n%v", err)
+	}
+	// The default is to have all persistent disks named linuxkit.vmdk
+	disk := devices.CreateDisk(controller, dss.Reference(), dss.Path(fmt.Sprintf("%s/%s", *newVM.vmName, "linuxkit.vmdk")))
+
+	disk.CapacityInKB = int64(newVM.persistentSz * 1024)
+
+	var add []types.BaseVirtualDevice
+	add = append(add, disk)
+
+	log.Infof("Adding a persistent disk to the Virtual Machine")
+
+	if vm.AddDevice(ctx, add...); err != nil {
+		log.Fatalf("Unable to add new storage device to VM configuration\n%v", err)
+	}
+}
+
+func addISO(ctx context.Context, newVM vmConfig, vm *object.VirtualMachine, dss *object.Datastore) {
+	devices, err := vm.Device(ctx)
+	if err != nil {
+		log.Fatalf("Unable to read devices from VM configuration\n%v", err)
+	}
+
+	ide, err := devices.FindIDEController("")
+	if err != nil {
+		log.Fatalf("Unable to find IDE device from VM configuration\n%v", err)
+	}
+
+	cdrom, err := devices.CreateCdrom(ide)
+	if err != nil {
+		log.Fatalf("Unable to create new CDROM device\n%v", err)
+	}
+
+	var add []types.BaseVirtualDevice
+	add = append(add, devices.InsertIso(cdrom, dss.Path(fmt.Sprintf("%s/%s", *newVM.vmName, "linuxkit.iso"))))
+
+	log.Infof("Adding ISO to the Virtual Machine")
+
+	if vm.AddDevice(ctx, add...); err != nil {
+		log.Fatalf("Unable to add new CD-ROM device to VM configuration\n%v", err)
+	}
+}
+
+func checkFile(file string) {
+	if _, err := os.Stat(file); err != nil {
+		if os.IsPermission(err) {
+			log.Fatalf("Unable to read file [%s], please check permissions", file)
+		}
+		if os.IsNotExist(err) {
+			log.Fatalf("File [%s], does not exist", file)
+		}
```

Comment:
can we add a catch all for a non-nil err that isn't one of these two cases?

ex: `log.Fatalf("Unable to stat file [%s]: %v", file, err)`